pulsar-broker-common/src/main/java/org/apache/pulsar/common/configuration/VipStatus.java

```diff
@@ -38,25 +44,75 @@ public class VipStatus {
     public static final String ATTRIBUTE_STATUS_FILE_PATH = "statusFilePath";
     public static final String ATTRIBUTE_IS_READY_PROBE = "isReadyProbe";
 
+    // log a full thread dump when a deadlock is detected in status check once every 10 minutes
+    // to prevent excessive logging
+    private static final long LOG_THREADDUMP_INTERVAL_WHEN_DEADLOCK_DETECTED = 600000L;
+    private static volatile long lastCheckStatusTimestamp;
```

Comment:
these shouldn't be static fields. This is a different case than the `BrokersBase` class where it was necessary.